lighty-core/lighty-codecs/src/test/java/io/lighty/codecs/DataCodecTest.java

```diff
@@ -128,6 +128,52 @@ public void testConvertNonexistingIdentifier() {
         dataCodec.convertIdentifier("/badToaster:badToaster");
     }
 
+    @Test
+    public void testCConvertIdentifierWithNoRevision() {
+        DataCodec<Toaster> dataCodec = new DataCodec<>(this.bindingCodecContext);
+        YangInstanceIdentifier expected = YangInstanceIdentifier.of(MAKE_TOAST_RPC_QNAME);
+        YangInstanceIdentifier ours = dataCodec.convertIdentifier("toaster:make-toast");
+        Assert.assertEquals(expected, ours);
+    }
+
+    @Test
+    public void testCConvertIdentifierWithCorrectRevision() {
+        DataCodec<Toaster> dataCodec = new DataCodec<>(this.bindingCodecContext);
+        YangInstanceIdentifier expected = YangInstanceIdentifier.of(MAKE_TOAST_RPC_QNAME);
+        YangInstanceIdentifier ours = dataCodec
+                .convertIdentifier("toaster@" + TOASTER_REVISION + ":make-toast");
+        Assert.assertEquals(expected, ours);
+    }
+
+    @Test
+    public void testCConvertIdentifierWithWonExistingRevision() {
+        DataCodec<Toaster> dataCodec = new DataCodec<>(this.bindingCodecContext);
+        YangInstanceIdentifier expected = YangInstanceIdentifier.of(MAKE_TOAST_RPC_QNAME);
+        YangInstanceIdentifier ours = dataCodec.convertIdentifier("toaster@1111-11-11:make-toast");
+        Assert.assertEquals(expected, ours);
+    }
+
+    @Test
+    public void testCConvertIdentifierNonExistingModule() {
+        DataCodec<Toaster> dataCodec = new DataCodec<>(this.bindingCodecContext);
+        Assert.assertThrows(IllegalStateException.class,
+            () -> dataCodec.convertIdentifier("toasterX:make-toast"));
+    }
+
+    @Test
+    public void testCConvertIdentifierWrongOrderIdentifier() {
+        DataCodec<Toaster> dataCodec = new DataCodec<>(this.bindingCodecContext);
+        Assert.assertThrows(IllegalStateException.class,
+            () -> dataCodec.convertIdentifier(TOASTER_REVISION + "@toaster:make-toast"));
+    }
+
+    @Test
+    public void testCConvertIdentifierWithWronglyFormattedRevision() {
+        DataCodec<Toaster> dataCodec = new DataCodec<>(this.bindingCodecContext);
+        Assert.assertThrows(IllegalStateException.class,
+            () -> dataCodec.convertIdentifier("toaster@err-11-11:make-toast"));
+    }
+
```

Comment:
Please, add more complicated tests too - to convert paths containing lists, keys, augmentations (different modules) and so on.